lib/event_listeners.js

```diff
@@ -281,6 +281,21 @@ AWS.EventListeners = {
       }
     });
 
+    addAsync('ERROR', 'error', function MAP_ERROR_CODE(err) {
+      var errorCodeMapping = this.service.api.errorCodeMapping;
+
+        if (errorCodeMapping) {
+          this.response.request.on('error', function (err, response) {
+            if (response.error && response.error.code) {
+              var mapping = errorCodeMapping[response.error.code];
+              if (mapping) {
+                response.error.code = mapping.code;
+              }
+            }
+          },true);
+        }
+    });
```

Comment:
```suggestion
    add('ERROR', 'error', function MAP_ERROR_CODE(resp) {
      var errorCodeMapping = resp.request.service.api.errorCodeMapping;
    
      if (errorCodeMapping && resp.error && resp.error.code) {
        var mapping = errorCodeMapping[response.error.code];
        if (mapping) {
          response.error.code = mapping.code;
        }
      }
    }, true);
```

Reply:
Since the event listener function is not an async function, you can just use `add` instead of `addAsync`